instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/InjectionState.java

```diff
@@ -0,0 +1,80 @@
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet;
+
+public class InjectionState {
+  private static final int ALREADY_INJECTED_FAKE_VALUE = -1;
+  private static final int HEAD_TAG_LENGTH = "<head>".length();
+  private int headTagBytesSeen = 0;
+  private String characterEncoding;
```

Comment:
does this work? (do we still need the setter after passing it into constructor)
```suggestion
  private final String characterEncoding;
```